src/NServiceBus.Metrics.ServiceControl/Buffers.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace NServiceBus.Metrics.ServiceControl
+{
+    using System;
+    using System.Threading;
+    using global::ServiceControl.Monitoring.Data;
+
+    class Buffers
+    {
+        public readonly RingBuffer ProcessingTime = new RingBuffer();
+        public readonly TaggedLongValueWriterV1 Writer = new TaggedLongValueWriterV1();
+
+        void Write(long value, string tag, RingBuffer buffer)
+        {
+            const int maxAttempts = 10;
+
+            if (buffer.TryWrite(value, Writer.GetTagId(tag)))
+            {
+                return;
+            }
+
+            var spin = new SpinWait();
+            for (var i = 0; i < maxAttempts; i++)
+            {
+                spin.SpinOnce();
+                if (buffer.TryWrite(value, Writer.GetTagId(tag)))
+                {
+                    return;
+                }
+            }
+        }
+
+        public void ReportProcessingTime(TimeSpan value, string messageType)
+        {
+            Write((long)value.TotalMilliseconds, messageType, ProcessingTime);
```

Comment:
I think that this should be getting reference to `Writer` as well. It's specific to `ProcessingTime` for example if we add `Retries` in the future tags collection for those two can we quite different.